src/utils/mardyn_assert.h

```diff
@@ -10,20 +10,25 @@
 
 #include "Logger.h"
 
-inline void mardyn_exit(int code) {
+// Macro to wrap mardyn_exit and pass the caller function, file and line
+#define MARDYN_EXIT(code) mardyn_exit(code, __FUNCTION__, __FILE__, __LINE__)
```

Comment:
Two things I slightly dislike about this line:
1. `__FUNCTION__` is afaik not part of the standard. If we use weird compilers this might cause trouble. The easy solution would be to leave it out (we have the file name and line. Should be enough). The fancy solution would be to check at compile time if the symbol is defined and then use it or not.
2. The content of `__FILE__` is relative to the working directory of the compiler at compile time. I would prefer to consistently only see the filename here, so a function should be added that extracts the basename.

Reply:
Addendum to 1.: The predefined variable [`__func__`](https://stackoverflow.com/questions/4384765/whats-the-difference-between-pretty-function-function-func) seems to be part of the C++11 standard.

Reply:
@HomesGH what about `__func__`?

Reply:
We could use `__func__` but do you think it is needed? We have the code location (file and line) and the error message.

Reply:
Might make it a bit more convenient to see the function name and get an idea about the nature of the error even without digging into the source code. But if this only clutters the output and makes it harder to read we can also leave it out.

Reply:
See this commit: https://github.com/ls1mardyn/ls1-mardyn/pull/341/commits/c80fde886adb60c1dc9a3387d2c4fc1ebb8317f0